server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java

```diff
@@ -1384,12 +1388,39 @@ public void setSessionTimeoutThreshold(int sessionTimeoutThreshold) {
     this.sessionTimeoutThreshold = sessionTimeoutThreshold;
   }
 
+  boolean isEnableReadOnly() {
+    return enableReadOnly;
+  }
+
+  void setEnableReadOnly(boolean enableReadOnly) {
+    this.enableReadOnly = enableReadOnly;
+  }
+
   public boolean isReadOnly() {
-    return readOnly;
+    return status == SystemStatus.READONLY;
   }
 
-  public void setReadOnly(boolean readOnly) {
-    this.readOnly = readOnly;
+  public SystemStatus getSystemStatus() {
+    return status;
+  }
+
+  public void setSystemStatus(SystemStatus newStatus) {
+    if (newStatus == SystemStatus.READONLY) {
+      if (enableReadOnly) {
+        logger.error(
+            "System mode will be set to {}! Only query statements are permitted!",
+            newStatus,
+            new RuntimeException("System mode is set to read-only"));
+      } else {
+        logger.error(
+            "Unrecoverable error occurs! Shutdown system directly when read-only mode is disabled.",
+            new RuntimeException("System mode is set to read-only"));
+        System.exit(-1);
```

Comment:
Do not shutdown system when disk is full.

Reply:
Add ReadOnlyStrategy enumeration and read_only_strategy parameter.